pkg/stack/loader.go

```diff
@@ -57,21 +60,41 @@ func LoadStacks(stackService Service) {
 			log.Fatalf("Failed to read stack: %s", err)
 		}
 
-		stackParameters := extractStackParameters(file)
+		hostnamePattern := extractMetadataParameters(file, hostnamePatternIdentifier)
+		if 0 < len(hostnamePattern) && len(hostnamePattern) < 2 {
+			stack.HostnamePattern = hostnamePattern[0]
+		}
+
+		hostnameVariable := extractMetadataParameters(file, hostnameVariableIdentifier)
+		if 0 < len(hostnameVariable) && len(hostnameVariable) < 2 {
+			stack.HostnameVariable = hostnameVariable[0]
+		}
+
+		err = stackService.Save(stack)
+		if err != nil {
+			log.Fatalf("Failed to update stack: %s", err)
+		}
+
+		consumedParameters := extractMetadataParameters(file, consumedParametersIdentifier)
+
+		stackParameters := extractMetadataParameters(file, stackParametersIdentifier)
 
 		requiredParameterSet := extractRequiredParameters(file, stackParameters)
 		fmt.Printf("Required parameters: %+v\n", requiredParameterSet)
-		for _, parameter := range requiredParameterSet {
-			_, err := stackService.CreateRequiredParameter(stack, parameter)
+		for _, name := range requiredParameterSet {
+			isConsumed := isConsumedParameter(name, consumedParameters)
+			_, err := stackService.CreateRequiredParameter(stack, name, isConsumed)
 			if err != nil {
 				log.Fatalf("Failed to create parameter: %s", err)
 			}
 		}
 
 		optionalParameterMap := extractOptionalParameters(file, stackParameters)
 		fmt.Printf("Optional parameters: %+v\n", optionalParameterMap)
-		for k, v := range optionalParameterMap {
-			_, err := stackService.CreateOptionalParameter(stack, k, v)
+		for name, v := range optionalParameterMap {
+			isConsumed := isConsumedParameter(name, consumedParameters)
+			log.Println("name: ", name, "v: ", v, "isConsumed: ", isConsumed)
```

Comment:
I would remove this line. Similar to the other log line I commented on: to me such logs are debug logs that might only highlight that we need some tests.